packages/widget/src/context/wallet.ts

```diff
@@ -54,14 +67,56 @@ export class WalletContextProvider extends BaseComponent {
   @provide({ context: walletContext })
   private walletContext: WalletContext = {};
 
+  @provide({ context: substrateProviderContext })
+  substrateProviderContext: SubstrateProviderContext = {};
+
   @property({ attribute: false, type: Object })
   evmWalllet?: EvmWallet;
 
-  connectedCallback(): void {
+  @property({ attribute: false })
+  substrateProviders?: Array<ApiPromise> = [];
+
+  /**
+   * Creates a provider map w.r.t parachain ids
+   * @param {Array<ApiPromise>} providers array of {@link ApiPromise}
+   * @returns {Promise<ParachainProviders>}
+   */
+  async createProvidersMap(
+    providers: Array<ApiPromise>
+  ): Promise<ParachainProviders> {
+    const map: ParachainProviders = new Map();
+
+    for (const provider of providers) {
+      try {
+        // ! following pallet might not be available on all chains
+        // ? should we add polkadot type augmentations here?
```

Comment:
ideally yes

Reply:
do we have a polkadot types package? i think we utilize that here @mpetrunic

Reply:
no idea, but we can install if needed